tools/recalc.py

```diff
@@ -214,11 +214,45 @@ async def recalculate_mode_scores(mode: GameMode, ctx: Context) -> None:
         await process_score_chunk(score_chunk, ctx)
 
 
+async def recalculate_score_status(mode: GameMode, ctx: Context) -> None:
+    pairs = await ctx.database.fetch_all(
+        "SELECT DISTINCT map_md5, userid, mode FROM scores WHERE status > 0 AND mode = :mode",
+        {"mode": mode},
+    )
+
+    for pair in pairs:
+        scores = await ctx.database.fetch_all(
+            "SELECT id, status, pp, play_time FROM scores "
+            "WHERE map_md5 = :map_md5 AND userid = :userid AND mode = :mode AND status > 0 "
+            "ORDER BY pp DESC",
+            pair,
+        )
+
+        best = sorted(scores, key=lambda x: (-x["pp"], x["play_time"]))[0]
+
+        if best["status"] != 2:
+            if DEBUG:
+                print(f"Status mismatch on score {best['id']}")
+
+            await ctx.database.execute(
+                "UPDATE scores SET status = 1 "
+                "WHERE map_md5 = :map_md5 AND userid = :userid AND mode = :mode AND status > 0",
+                pair,
+            )
+
+            await ctx.database.execute(
+                "UPDATE scores SET status = 2 WHERE id = :id",
+                {"id": best["id"]},
+            )
```

Comment:
ideally this runs in a transaction, if an API request comes in inbetween these 2 queries or something then they're going to have *no* best on the map.

additionally, for no reason other than being picky, would be good to add a `AND id != :best_id` here just so that doesn't get unnecessarily touched until the following query